src/topicFactory.js

```diff
@@ -3,91 +3,100 @@ import qs from 'qs'
+        window.onhashchange = () => {
+          if (window.location.hash) {
+            const params = qs.parse(window.location.hash.slice(1))
+            if (params[topicName]) {
+              window.onhashchange = () => {}
+              window.location.hash = ''
+              cb(null, params[topicName])
+            } else if (params.error) {
+              window.onhashchange = () => {}
+              window.location.hash = ''
+              cb(params.error)
             }
-          } catch (err) {
-            console.error(err.stack)
-            clearInterval(interval)
-            return cb(err)
           }
-          // Check for param, stop polling and callback if present
-          if (data && data[topicName]) {
-            clearInterval(interval)
-            clearTopic(url)
-            return cb(null, data[topicName])
-          }
-        })
-      }, pollingInterval)
-  }
+        }
+      }
 
+      const pollForResult = (topicName, url, cb) => {
+        let interval = setInterval(
 
-  function clearTopic (url) {
-    nets({
-      uri: url,
-      method: 'DELETE',
-      rejectUnauthorized: false
-    }, function (err) { if (err) { throw err } /* Errors withouth this cb */ })
-  }
+          () => {
+            nets({
+              uri: url,
+              method: 'GET',
+              rejectUnauthorized: false
+            },
+
+           (err, res, body) => {
+              if (err) return cb(err)
+              // TODO what is good error message
+              if (stopBool) return cb(new Error('Request Cancelled'))
+              // parse response into raw account
+              let data
 
-  function newTopic (topicName) {
-    let url
-    if (isOnMobile) {
-      url = window.location.href
-    } else {
-      url = chasquiUrl + randomString(16)
-    }
-    const topic = new Promise((resolve, reject) => {
-      const cb = (error, response) => {
-        if (error) return reject(error)
-        resolve(response)
+              try {
+                data = JSON.parse(body).message
+                if (data.error) {
+                  clearInterval(interval)
+                  return cb(data.error)
+                }
+              } catch (err) {
+                console.error(err.stack)
+                clearInterval(interval)
+                return cb(err)
+              }
+              // Check for param, stop polling and callback if present
+              if (data && data[topicName]) {
+                clearInterval(interval)
+                clearTopic(url)
+                return cb(null, data[topicName])
+              }
+            })
+          }, pollingInterval)
       }
-      if (isOnMobile) {
-        waitForHashChange(topicName, cb)
-      } else {
-        pollForResult(topicName, url, cb)
+
+      return {
+        stop: () => {
+          stopBool = true
+        },
+        url: url,
+        stopBool: stopBool,
+        listen:  () => (
```

Comment:
yes ideally, I agree, but after trying many options, this was minimal change I could find. I had originally attempted adding it as  func to the topic object, but this stop state has be wrapped in a closure where the promise can internally reference it, by attaching it to the promise object it, it cannot be referenced with in the func passed to the promise. I can revisit it to provide a more specific example, but I did try this. The tradeoffs were also minimal, only the topic api changed, which is mostly wrapped in our other stuff anyways and this cancel ability is necessary. Another option is add a promise library the allows canceling promises.